crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -319,17 +334,16 @@ impl<'a> Submitter<'a> {
                 return SubmissionError::from(err);
             }
 
-            // if gas price has increased cancel old transaction so new transaction could be submitted.
-
-            if let Some((previous_tx, previous_gas_price)) = transactions.last() {
+            // if gas price has not increased enough, skip submitting the transaction.
+            if let Some(previous_gas_price) = transactions
+                .last()
+                .map(|(_, previous_gas_price)| previous_gas_price)
+                .or_else(|| pending_gas_price.as_ref())
```

Comment:
Is it correct that we only ever use `pending_gas_price` on the first iteration of the loop? On subsequent iterations `transactions.last()` would probably yield a value from which we can take the gas price, right?

Also since it's possible to return from this function before we reach this line, could we do something like `.or_else(calculate_pending_gas_price)` instead of `.or_else(|| pre_calculated_pending_gas_price)`. This way we could delay a possibly (I don't know) expensive operation until we really need it.
Not sure if it's worth the effort though, given the fact that the reasons to return before we reach this line seem quite exotic.

Reply:
It is possible to use `pending_gas_price` more then just in first iteration, because we have a `continue` between usage of `pending_gas_price` and submitting the transaction (after which `transactions.last()` yields something).

Reply:
Ahh, I see. In that case trying to delay computing the pending gas cost and caching that result doesn't seem to be worth the added complexity.